File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        String vowels = "aeiouAEIOU";
        ArrayList<Character> resList = new ArrayList<>();
        ArrayList<Integer> index = new ArrayList<>();

        for (int i = 0; i < s.length(); i++) {
            if (vowels.indexOf(s.charAt(i)) != -1) {
                resList.add(s.charAt(i));
                index.add(i);
            }
        }

        Collections.sort(resList);

        StringBuilder result = new StringBuilder(s);

        for (int i = 0; i < index.size(); i++) {
            result.setCharAt(index.get(i), resList.get(i));
        }

        return result.toString();
    }
}

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowel_string = 'AEIOUaeiou'
        vowel_list = []
        res = ''
        for ch in s:
            if ch in vowel_string:
                vowel_list.append(ch)
            
        new_list = sorted(vowel_list)
        i = 0
        for ch in s:
            if ch in vowel_string and i < len(new_list):
                res += new_list[i]
                i+=1
            else:
                res += ch

        return res        

